### PR TITLE
Updated packages to support node 16

### PR DESCRIPTION
**Pull request by jez500**
Doing some testing on this atm, but this will raise minimum node version to 14+ (built on node 16)

Most notable change is getting rid of `node-sass` in favour of `sass` which is a pure JS implementation. Node sass has been deprecated https://sass-lang.com/blog/libsass-is-deprecated

Have already pushed to npm for testing https://www.npmjs.com/package/@doghouse/webpack-base, once testing complete will merge + tag this accordingly

To use this, a project `package.json` will need to update deps with something like
```
  "devDependencies": {
    "@doghouse/webpack-base": "^1.16.0"
  }
```

### CODE REVIEW

File: packages/webpack-base/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+This is a helper package that abstracts common webpack configuration and plugins. It allows adding custom config
+which gets merged with the base config using [Webpack merge](https://github.com/survivejs/webpack-merge).
+
+Includes common packages required by most projects such as Babel, SASS, Webpack, etc. See [package.json](package.json)
+for a full list
+
+## Versions
+
+Going forward, versioning aligns with nodejs version requirements. The second part of the version should align with 
+tested node version, eg `1.NODE_VERSION.0`
```
Comment (damyon) [inline packages/webpack-base/README.md:12]: +1 for adding this to the version info.

**Comment by damyon** [comment]
Not saying we should do this here - but I integrated this https://github.com/leafo/scssphp php SCSS compiler into Moodle (no webpack).

**Comment by jez500** [comment]
> Not saying we should do this here - but I integrated this https://github.com/leafo/scssphp php SCSS compiler into Moodle (no webpack).

interesting!

**Comment by jez500** [comment]
Note with this, we start getting deprecation errors, still works but will have to address them at some point
```
DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($r * 299 + $g * 587 + $b * 114, 1000)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
36 │   $yiq: (($r * 299) + ($g * 587) + ($b * 114)) / 1000;
   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   ╵
    node_modules/@doghouse/uikit/scss/tools/_functions.scss 36:9  duk-color-yiq()
    node_modules/@doghouse/uikit/scss/base/_wysiwyg.scss 7:17     @content
    node_modules/@doghouse/uikit/scss/tools/_wysiwyg.scss 14:5    duk-wysiwyg-only()
    node_modules/@doghouse/uikit/scss/base/_wysiwyg.scss 6:3      @import
    node_modules/@doghouse/uikit/scss/_duk.scss 25:9              @import
    stdin 32:9                                                    root stylesheet

WARNING: You probably don't mean to use the color value transparent in interpolation here.
It may end up represented as transparent, which will likely produce invalid CSS.
Always quote color names when using them as strings or map keys (for example, "transparent").
If you really want to use the color value here, use '"" + $button-variant'.

   ╷
26 │   .button--#{$button-variant},
   │              ^^^^^^^^^^^^^^^
   ╵
    docroot/themes/contrib/district_base/styles/scss/objects/_button.scss 26:14  @import
    stdin 47:9      
```

**Pull request merged**